src/rpc_client/src/meta_client.rs

```diff
@@ -1221,17 +1224,57 @@ impl GrpcMetaClient {
             .map_err(RpcError::TransportError)
     }
 
-    pub(crate) async fn try_build_rpc_channel(addrs: Vec<String>) -> Result<(Channel, String)> {
+    fn get_retry_strategy() -> impl Iterator<Item = Duration> {
+        ExponentialBackoff::from_millis(Self::CONN_RETRY_BASE_INTERVAL_MS)
+            .max_delay(Duration::from_millis(Self::CONN_RETRY_MAX_INTERVAL_MS))
+            .map(jitter)
+    }
+
+    /// Connect against one of multiple meta nodes using a K8s service
+    pub(crate) async fn try_build_rpc_channel_service(addr: String) -> Result<(Channel, String)> {
+        let endpoint = Self::addr_to_endpoint(addr.clone())?;
+
+        let res = tokio_retry::Retry::spawn(Self::get_retry_strategy(), || async {
+            let endpoint = endpoint.clone();
+            let addr = addr.clone();
+            match Self::connect_to_endpoint(endpoint).await {
+                Ok(channel) => {
+                    tracing::info!(
+                        "Connect to meta server via service at {} successfully",
+                        addr
+                    );
+                    Ok((channel, addr))
+                }
+                Err(e) => {
+                    tracing::warn!(
+                        "Failed to connect to meta server using service at {}, trying again: {}",
+                        addr,
+                        e
+                    );
+                    Err(())
+                }
+            }
+        })
+        .await;
+
+        match res {
```

Comment:
The `res` will always be `Ok` because the retry strategy will ensure retry connecting to endpoint until success.

Reply:
Are you sure? `CONN_RETRY_MAX_INTERVAL_MS` is equal to 5000. Retrying for 5 seconds in a K8s environment does not seem very long to me, especially if we have network issues.

Reply:
Yes, `CONN_RETRY_MAX_INTERVAL_MS` is just the max delay of interval between two retries. Actually the error will only be used to trigger retry in next round.

Reply:
Okay. If we retry then I am fine with it. Just changed the logging slightly.